### PR TITLE
Opmphm Predictor update + some benchmark update

### PR DESCRIPTION
**Pull request by KurtMi**
## Basics

Do not describe the purpose here but:

- [x] Short descriptions should be in the release notes (added as entry in
      doc/news/_preparation_next_release.md which contains `*(my name)*`)
      **Please always add something to the the release notes.**
- [ ] Longer descriptions should be in documentation or in design decisions.
- [x] Describe details of how you changed the code in commit messages
      (first line should have `module: short statement` syntax)
- [ ] References to issues, e.g. `close #X`, should be in the commit messages.

## Checklist

Check relevant points but **please do not remove entries**.
For docu fixes, spell checking, and similar none of these points below
need to be checked.

- [x] I added unit tests
- [x] I ran all tests locally and everything went fine
- [x] affected documentation is fixed
- [x] I added code comments, logging, and assertions (see doc/CODING.md)
- [ ] meta data is updated (e.g. README.md of plugins)

## Review

Remove the line below and add the "work in progress" label if you do
not want the PR to be reviewed:

### CODE REVIEW

**Comment by KurtMi** [comment]
jenkins build all please

**Comment by KurtMi** [comment]
jenkins build all please

File: benchmarks/opmphm.c [hunk 1]
```diff
@@ -1124,9 +1124,13 @@ void benchmarkOPMPHMBuildTime (char * name)
 
 	printf ("Run Benchmark %s:\n", name);
 
-	// for all KeySet shapes
+	// for all KeySet shapes except 6
```
Comment (markus2330) [inline benchmarks/opmphm.c]: Could you please explain this exception in the comment?

File: doc/news/_preparation_next_release.md [hunk 2]
```diff
@@ -210,8 +210,8 @@ Thanks to Daniel Bugl.
 
 ### Core
 
-- The KeySet has a new flag `KS_FLAG_REBUILD_OPMPHM`, this flag will be used by the hybrid search
-  that combines binary search and [OPMPHM](https://master.libelektra.org/doc/dev/data-structures.md#order-preserving-minimal-perfect-hash-map-aka-opmphm) *(Kurt Micheli)*
+- The KeySet has a new flag `KS_FLAG_NAME_CHANGE`, this flag will be used by the hybrid search
```
Comment (markus2330) [inline doc/news/_preparation_next_release.md]: What does the flag do? And isn't it a rename of an existing flag?

Please move text to a section "Proposal", it is not yet part of Core.

File: src/include/kdbopmphmpredictor.h [hunk 3]
```diff
@@ -37,6 +37,7 @@ typedef struct
 	uint8_t * patternTable; /*!< the global pattern history table */
 	size_t size;		/*!< size of patternTable in bytes */
 	size_t lookupCount;     /*!< number of lookups made without alteration of the KeySet */
+	size_t ksSize;		/*!< number of keys in the KeySet */
```
Comment (markus2330) [inline src/include/kdbopmphmpredictor.h]: What is different here to the size of the keyset?

File: src/include/kdbprivate.h [hunk 4]
```diff
@@ -139,7 +139,7 @@ typedef enum {
 		 this flag will be set, so that the backend will sync
 		 the keys to database.*/
 #ifdef ELEKTRA_ENABLE_OPTIMIZATIONS
-	,KS_FLAG_REBUILD_OPMPHM = 1 << 1 /*!<
+	,KS_FLAG_NAME_CHANGE = 1 << 1 /*!<
```
Comment (markus2330) [inline src/include/kdbprivate.h]: Why is the new name better?

File: src/include/kdbopmphmpredictor.h (hunk 3 above)
Comment (KurtMi) [reply src/include/kdbopmphmpredictor.h]: The predictor generally looks at old events to predict actual once.
At the prediction the keyset size is need to determine if it **was** worth using the OPMPHM or not. But at the prediction the keyset size may has changed, therefore the old one is needed.

File: doc/news/_preparation_next_release.md (hunk 2 above)
Comment (KurtMi) [reply doc/news/_preparation_next_release.md]: A keyset is marked with this flag when a key name has changed. It is a existing one but not jet used.

File: src/include/kdbopmphmpredictor.h (hunk 3 above)
Comment (markus2330) [reply src/include/kdbopmphmpredictor.h]: Thank you. This information should be in the comment.

**Comment by KurtMi** [comment]
jenkins build all please

**Comment by KurtMi** [comment]
jenkins build all please

**Comment by KurtMi** [comment]
@markus2330 I made the requested changes! My thesis will explain all in detail!

**Comment by markus2330** [comment]
Thank you, good job! I am looking forward to reading it!

**Pull request merged**